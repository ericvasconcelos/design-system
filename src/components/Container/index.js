import React from 'react'

import { node, string } from 'prop-types'

import { Wrapper } from './styles'

export const Container = ({ children, className }) => (
  <Wrapper data-testid="Container" className={className}>
    {children}
  </Wrapper>
)

Container.propTypes = {
  children: node.isRequired,
  className: string
}

Container.defaultProps = {
  className: ''
}
