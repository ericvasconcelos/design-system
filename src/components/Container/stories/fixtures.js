import styled from 'styled-components'

export const Content = styled.div`
  width: 100%;
  height: 100px;
  background: ${({ theme }) => theme.pallete.greyCashme[300]};
  display: flex;
  justify-content: center;
  align-items: center;
`
