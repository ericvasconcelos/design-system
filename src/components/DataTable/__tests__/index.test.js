import React from 'react'

import userEvent from '@testing-library/user-event'
import { DataTable } from 'components/DataTable'
import { styledRender } from 'utils/styledRender'

import { headers, data, collapse, fourItems } from './fixtures'

describe('Render DataTable Component when', () => {
  const onChangePage = jest.fn()
  const onSort = jest.fn()

  it('set perPage and have exact 1 item left', () => {
    const { getByText } = styledRender(
      <DataTable
        headers={headers}
        data={fourItems}
        totalRows={5}
        currentPage={1}
        numberPages={2}
        perPage={4}
        onChangePage={onChangePage}
        onSort={onSort}
        showMoreItemsText="Mostrar mais (1)"
      />
    )

    expect(getByText('Mostrar mais (1)')).toBeInTheDocument()
  })

  it('have common values', () => {
    const { container, getByText, getAllByText } = styledRender(
      <DataTable
        headers={headers}
        data={data}
        totalRows={10}
        currentPage={1}
        numberPages={6}
        onChangePage={onChangePage}
        onSort={onSort}
      />
    )


    expect(getAllByText('Label')).toHaveLength(10)
    expect(getByText('Header 1')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('have collapse values', () => {
    const { container, getAllByText } = styledRender(
      <DataTable
        headers={headers}
        data={data}
        totalRows={10}
        currentPage={1}
        numberPages={6}
        onChangePage={onChangePage}
        onSort={onSort}
        collapse={collapse}
      />
    )

    expect(getAllByText('Label')).toHaveLength(10)
    expect(getAllByText('Header 5')[0]).not.toBeVisible()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('click collapse button open new row', () => {
    const { getAllByText, getAllByRole } = styledRender(
      <DataTable
        headers={headers}
        data={data}
        totalRows={10}
        currentPage={1}
        numberPages={6}
        onChangePage={onChangePage}
        onSort={onSort}
        collapse={collapse}
      />
    )
    const buttons = getAllByRole('button', { name: 'expandir linha' })
    const rowsCollapse = getAllByText('Header 5')

    userEvent.click(buttons[0])

    expect(buttons).toHaveLength(10)
    expect(rowsCollapse[0]).toBeVisible()
  })

  it('click sortable header call callback', () => {
    const { getByText } = styledRender(
      <DataTable
        headers={headers}
        data={data}
        totalRows={10}
        currentPage={1}
        numberPages={6}
        onChangePage={onChangePage}
        onSort={onSort}
      />
    )
    const headerSortable = getByText('Header 2')

    userEvent.click(headerSortable)

    expect(onSort).toHaveBeenCalledWith('header2')
  })

  it('click show more call callback', () => {
    const { getByRole } = styledRender(
      <DataTable
        headers={headers}
        data={data}
        totalRows={10}
        currentPage={1}
        numberPages={10}
        onChangePage={onChangePage}
        onSort={onSort}
      />
    )
    const showMoreButton = getByRole('button')

    userEvent.click(showMoreButton)

    expect(onChangePage).toHaveBeenCalledWith(2)
  })

  it('have one page dont render pagination', () => {
    const { queryByRole } = styledRender(
      <DataTable
        headers={headers}
        data={data}
        totalRows={10}
        currentPage={1}
        numberPages={1}
        onChangePage={onChangePage}
        onSort={onSort}
      />
    )
    const showMoreButton = queryByRole('button')

    expect(showMoreButton).not.toBeInTheDocument()
  })

  it('its the last page dont render pagination', () => {
    const { queryByRole } = styledRender(
      <DataTable
        headers={headers}
        data={data}
        totalRows={10}
        currentPage={10}
        numberPages={10}
        onChangePage={onChangePage}
        onSort={onSort}
      />
    )
    const showMoreButton = queryByRole('button')

    expect(showMoreButton).not.toBeInTheDocument()
  })

  it('have less that 10 rest items render exact number', () => {
    const { getByText } = styledRender(
      <DataTable
        headers={headers}
        data={data}
        totalRows={15}
        currentPage={1}
        numberPages={2}
        onChangePage={onChangePage}
        onSort={onSort}
        perPage={5}
        showMoreItemsText="Mostrar mais (5)"
      />
    )

    expect(getByText('Mostrar mais (5)')).toBeInTheDocument()
  })

  it('have one only rest item render exact number', () => {
    const { getByText } = styledRender(
      <DataTable
        headers={headers}
        data={data}
        totalRows={11}
        currentPage={1}
        numberPages={2}
        onChangePage={onChangePage}
        onSort={onSort}
        perPage={1}
        showMoreItemsText="Mostrar mais 1"
      />
    )

    expect(getByText('Mostrar mais 1')).toBeInTheDocument()
  })
})
