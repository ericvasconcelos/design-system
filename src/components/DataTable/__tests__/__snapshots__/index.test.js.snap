// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render DataTable Component when have collapse values 1`] = `
.c5 {
  -webkit-mask-image: url(https://qa-static.cashme.com.br/icons/chevron-down.svg);
  mask-image: url(https://qa-static.cashme.com.br/icons/chevron-down.svg);
  background-color: #00224B;
  background-image: none;
  background-size: 24px;
  width: 24px;
  height: 24px;
  -webkit-mask-position-x: 0;
  mask-position-x: 0;
  -webkit-mask-size: 24px;
  mask-size: 24px;
  -webkit-mask-repeat: no-repeat;
  mask-repeat: no-repeat;
}

.c11 {
  -webkit-mask-image: url(https://qa-static.cashme.com.br/icons/chevron-down.svg);
  mask-image: url(https://qa-static.cashme.com.br/icons/chevron-down.svg);
  background-color: #69788C;
  background-image: none;
  background-size: 24px;
  width: 24px;
  height: 24px;
  -webkit-mask-position-x: 0;
  mask-position-x: 0;
  -webkit-mask-size: 24px;
  mask-size: 24px;
  -webkit-mask-repeat: no-repeat;
  mask-repeat: no-repeat;
}

.c1 {
  border-collapse: separate;
  width: 100%;
  margin: auto;
  border-spacing: 0 0.5rem;
}

.c1 td {
  color: none;
  font-size: 0.9375rem;
  font-weight: 400;
  line-height: 1.219rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

.c1 td:first-child,
.c1 th:first-child {
  border-radius: 8px 0 0 8px;
}

.c1 td:last-child,
.c1 th:last-child {
  border-radius: 0 8px 8px 0;
}

.c2 {
  padding: 1rem;
  position: relative;
  cursor: auto;
}

.c4 {
  padding: 1rem;
  position: relative;
  cursor: pointer;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
  white-space: nowrap;
  text-align: left;
  color: #00224B;
  font-size: 0.9375rem;
  font-weight: 500;
  line-height: 1.219rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

.c6 {
  width: 16px;
  height: 16px;
  -webkit-mask-size: 16px;
  mask-size: 16px;
  background: #00224B;
}

.c8 {
  padding: 1rem;
  text-align: left;
  border: 1px solid transparent;
  color: #00224B;
  font-size: 0.9375rem;
  font-weight: 400;
  line-height: 1.219rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

.c8:focus {
  border: 1px solid #8DC6FF;
  box-shadow: 0 0 0 4px #003FF23D;
}

.c9 {
  cursor: pointer;
  padding: 0;
  vertical-align: middle;
  line-height: 0;
  max-width: 80px;
}

.c7 {
  border-bottom: 1px solid rgba(0,0,0,0.08);
  display: table-row;
  background-color: #FFFFFF;
  border-top-left-radius: 10px;
}

.c12 {
  border-bottom: 1px solid rgba(0,0,0,0.08);
  display: none;
  background-color: #FFFFFF;
  border-top-left-radius: 10px;
}

.c10 {
  cursor: pointer;
}

@media (max-width:1024px) {
  .c0 {
    overflow-x: auto;
  }
}

<div
  class="c0"
  data-testid="DataTable"
>
  <table
    aria-rowcount="10"
    class="c1"
    role="grid"
  >
    <thead>
      <tr
        role="row"
      >
        <th
          class="c2"
          id="header-header1"
          role="columnheader"
        >
          <div
            class="c3"
          >
            Header 1
          </div>
        </th>
        <th
          class="c4"
          id="header-header2"
          role="columnheader"
        >
          <div
            class="c3"
          >
            Header 2
            <div
              aria-hidden="true"
              aria-label=""
              class="c5 c6"
              data-testid="Icon"
            />
          </div>
        </th>
        <th
          class="c4"
          id="header-header3"
          role="columnheader"
        >
          <div
            class="c3"
          >
            Header 3
            <div
              aria-hidden="true"
              aria-label=""
              class="c5 c6"
              data-testid="Icon"
            />
          </div>
        </th>
        <th
          class="c2"
          id="header-header4"
          role="columnheader"
        >
          <div
            class="c3"
          >
            Header 4
          </div>
        </th>
        <th
          class="c2"
        />
      </tr>
    </thead>
    <tbody>
      <tr
        aria-rowindex="0"
        class="c7"
        role="row"
      >
        <td
          class="c8"
          headers="header-header1"
          role="cell"
          tabindex="0"
        >
          lorem ipsum
        </td>
        <td
          class="c8"
          headers="header-header2"
          role="cell"
          tabindex="-1"
        >
          R$ 1000
        </td>
        <td
          class="c8"
          headers="header-header3"
          role="cell"
          tabindex="-1"
        >
          Label
        </td>
        <td
          class="c8"
          headers="header-header4"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8 c9"
          role="cell"
          tabindex="-1"
        >
          <button
            class="c10"
            tabindex="-1"
            type="button"
          >
            <div
              aria-hidden="false"
              aria-label="expandir linha"
              class="c11"
              data-testid="Icon"
              role="button"
            />
          </button>
        </td>
      </tr>
      <tr
        aria-rowindex="0"
        class="c12"
        role="row"
      >
        <td
          class="c8"
          colspan="5"
        >
          <div>
            <p>
              <strong>
                Header 5
              </strong>
            </p>
            <p>
              Lorem Ipsum
            </p>
          </div>
        </td>
      </tr>
      <tr
        aria-rowindex="1"
        class="c7"
        role="row"
      >
        <td
          class="c8"
          headers="header-header1"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8"
          headers="header-header2"
          role="cell"
          tabindex="-1"
        >
          R$ 1000
        </td>
        <td
          class="c8"
          headers="header-header3"
          role="cell"
          tabindex="-1"
        >
          Label
        </td>
        <td
          class="c8"
          headers="header-header4"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8 c9"
          role="cell"
          tabindex="-1"
        >
          <button
            class="c10"
            tabindex="-1"
            type="button"
          >
            <div
              aria-hidden="false"
              aria-label="expandir linha"
              class="c11"
              data-testid="Icon"
              role="button"
            />
          </button>
        </td>
      </tr>
      <tr
        aria-rowindex="1"
        class="c12"
        role="row"
      >
        <td
          class="c8"
          colspan="5"
        >
          <div>
            <p>
              <strong>
                Header 5
              </strong>
            </p>
            <p>
              Lorem Ipsum
            </p>
          </div>
        </td>
      </tr>
      <tr
        aria-rowindex="2"
        class="c7"
        role="row"
      >
        <td
          class="c8"
          headers="header-header1"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8"
          headers="header-header2"
          role="cell"
          tabindex="-1"
        >
          R$ 1000
        </td>
        <td
          class="c8"
          headers="header-header3"
          role="cell"
          tabindex="-1"
        >
          Label
        </td>
        <td
          class="c8"
          headers="header-header4"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8 c9"
          role="cell"
          tabindex="-1"
        >
          <button
            class="c10"
            tabindex="-1"
            type="button"
          >
            <div
              aria-hidden="false"
              aria-label="expandir linha"
              class="c11"
              data-testid="Icon"
              role="button"
            />
          </button>
        </td>
      </tr>
      <tr
        aria-rowindex="2"
        class="c12"
        role="row"
      >
        <td
          class="c8"
          colspan="5"
        >
          <div>
            <p>
              <strong>
                Header 5
              </strong>
            </p>
            <p>
              Lorem Ipsum
            </p>
          </div>
        </td>
      </tr>
      <tr
        aria-rowindex="3"
        class="c7"
        role="row"
      >
        <td
          class="c8"
          headers="header-header1"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8"
          headers="header-header2"
          role="cell"
          tabindex="-1"
        >
          R$ 1000
        </td>
        <td
          class="c8"
          headers="header-header3"
          role="cell"
          tabindex="-1"
        >
          Label
        </td>
        <td
          class="c8"
          headers="header-header4"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8 c9"
          role="cell"
          tabindex="-1"
        >
          <button
            class="c10"
            tabindex="-1"
            type="button"
          >
            <div
              aria-hidden="false"
              aria-label="expandir linha"
              class="c11"
              data-testid="Icon"
              role="button"
            />
          </button>
        </td>
      </tr>
      <tr
        aria-rowindex="3"
        class="c12"
        role="row"
      >
        <td
          class="c8"
          colspan="5"
        >
          <div>
            <p>
              <strong>
                Header 5
              </strong>
            </p>
            <p>
              Lorem Ipsum
            </p>
          </div>
        </td>
      </tr>
      <tr
        aria-rowindex="4"
        class="c7"
        role="row"
      >
        <td
          class="c8"
          headers="header-header1"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8"
          headers="header-header2"
          role="cell"
          tabindex="-1"
        >
          R$ 1000
        </td>
        <td
          class="c8"
          headers="header-header3"
          role="cell"
          tabindex="-1"
        >
          Label
        </td>
        <td
          class="c8"
          headers="header-header4"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8 c9"
          role="cell"
          tabindex="-1"
        >
          <button
            class="c10"
            tabindex="-1"
            type="button"
          >
            <div
              aria-hidden="false"
              aria-label="expandir linha"
              class="c11"
              data-testid="Icon"
              role="button"
            />
          </button>
        </td>
      </tr>
      <tr
        aria-rowindex="4"
        class="c12"
        role="row"
      >
        <td
          class="c8"
          colspan="5"
        >
          <div>
            <p>
              <strong>
                Header 5
              </strong>
            </p>
            <p>
              Lorem Ipsum
            </p>
          </div>
        </td>
      </tr>
      <tr
        aria-rowindex="5"
        class="c7"
        role="row"
      >
        <td
          class="c8"
          headers="header-header1"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8"
          headers="header-header2"
          role="cell"
          tabindex="-1"
        >
          R$ 1000
        </td>
        <td
          class="c8"
          headers="header-header3"
          role="cell"
          tabindex="-1"
        >
          Label
        </td>
        <td
          class="c8"
          headers="header-header4"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8 c9"
          role="cell"
          tabindex="-1"
        >
          <button
            class="c10"
            tabindex="-1"
            type="button"
          >
            <div
              aria-hidden="false"
              aria-label="expandir linha"
              class="c11"
              data-testid="Icon"
              role="button"
            />
          </button>
        </td>
      </tr>
      <tr
        aria-rowindex="5"
        class="c12"
        role="row"
      >
        <td
          class="c8"
          colspan="5"
        >
          <div>
            <p>
              <strong>
                Header 5
              </strong>
            </p>
            <p>
              Lorem Ipsum
            </p>
          </div>
        </td>
      </tr>
      <tr
        aria-rowindex="6"
        class="c7"
        role="row"
      >
        <td
          class="c8"
          headers="header-header1"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8"
          headers="header-header2"
          role="cell"
          tabindex="-1"
        >
          R$ 1000
        </td>
        <td
          class="c8"
          headers="header-header3"
          role="cell"
          tabindex="-1"
        >
          Label
        </td>
        <td
          class="c8"
          headers="header-header4"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8 c9"
          role="cell"
          tabindex="-1"
        >
          <button
            class="c10"
            tabindex="-1"
            type="button"
          >
            <div
              aria-hidden="false"
              aria-label="expandir linha"
              class="c11"
              data-testid="Icon"
              role="button"
            />
          </button>
        </td>
      </tr>
      <tr
        aria-rowindex="6"
        class="c12"
        role="row"
      >
        <td
          class="c8"
          colspan="5"
        >
          <div>
            <p>
              <strong>
                Header 5
              </strong>
            </p>
            <p>
              Lorem Ipsum
            </p>
          </div>
        </td>
      </tr>
      <tr
        aria-rowindex="7"
        class="c7"
        role="row"
      >
        <td
          class="c8"
          headers="header-header1"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8"
          headers="header-header2"
          role="cell"
          tabindex="-1"
        >
          R$ 1000
        </td>
        <td
          class="c8"
          headers="header-header3"
          role="cell"
          tabindex="-1"
        >
          Label
        </td>
        <td
          class="c8"
          headers="header-header4"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8 c9"
          role="cell"
          tabindex="-1"
        >
          <button
            class="c10"
            tabindex="-1"
            type="button"
          >
            <div
              aria-hidden="false"
              aria-label="expandir linha"
              class="c11"
              data-testid="Icon"
              role="button"
            />
          </button>
        </td>
      </tr>
      <tr
        aria-rowindex="7"
        class="c12"
        role="row"
      >
        <td
          class="c8"
          colspan="5"
        >
          <div>
            <p>
              <strong>
                Header 5
              </strong>
            </p>
            <p>
              Lorem Ipsum
            </p>
          </div>
        </td>
      </tr>
      <tr
        aria-rowindex="8"
        class="c7"
        role="row"
      >
        <td
          class="c8"
          headers="header-header1"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8"
          headers="header-header2"
          role="cell"
          tabindex="-1"
        >
          R$ 1000
        </td>
        <td
          class="c8"
          headers="header-header3"
          role="cell"
          tabindex="-1"
        >
          Label
        </td>
        <td
          class="c8"
          headers="header-header4"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8 c9"
          role="cell"
          tabindex="-1"
        >
          <button
            class="c10"
            tabindex="-1"
            type="button"
          >
            <div
              aria-hidden="false"
              aria-label="expandir linha"
              class="c11"
              data-testid="Icon"
              role="button"
            />
          </button>
        </td>
      </tr>
      <tr
        aria-rowindex="8"
        class="c12"
        role="row"
      >
        <td
          class="c8"
          colspan="5"
        >
          <div>
            <p>
              <strong>
                Header 5
              </strong>
            </p>
            <p>
              Lorem Ipsum
            </p>
          </div>
        </td>
      </tr>
      <tr
        aria-rowindex="9"
        class="c7"
        role="row"
      >
        <td
          class="c8"
          headers="header-header1"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8"
          headers="header-header2"
          role="cell"
          tabindex="-1"
        >
          R$ 1000
        </td>
        <td
          class="c8"
          headers="header-header3"
          role="cell"
          tabindex="-1"
        >
          Label
        </td>
        <td
          class="c8"
          headers="header-header4"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8 c9"
          role="cell"
          tabindex="-1"
        >
          <button
            class="c10"
            tabindex="-1"
            type="button"
          >
            <div
              aria-hidden="false"
              aria-label="expandir linha"
              class="c11"
              data-testid="Icon"
              role="button"
            />
          </button>
        </td>
      </tr>
      <tr
        aria-rowindex="9"
        class="c12"
        role="row"
      >
        <td
          class="c8"
          colspan="5"
        >
          <div>
            <p>
              <strong>
                Header 5
              </strong>
            </p>
            <p>
              Lorem Ipsum
            </p>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Render DataTable Component when have common values 1`] = `
.c5 {
  -webkit-mask-image: url(https://qa-static.cashme.com.br/icons/chevron-down.svg);
  mask-image: url(https://qa-static.cashme.com.br/icons/chevron-down.svg);
  background-color: #00224B;
  background-image: none;
  background-size: 24px;
  width: 24px;
  height: 24px;
  -webkit-mask-position-x: 0;
  mask-position-x: 0;
  -webkit-mask-size: 24px;
  mask-size: 24px;
  -webkit-mask-repeat: no-repeat;
  mask-repeat: no-repeat;
}

.c1 {
  border-collapse: separate;
  width: 100%;
  margin: auto;
  border-spacing: 0 0.5rem;
}

.c1 td {
  color: none;
  font-size: 0.9375rem;
  font-weight: 400;
  line-height: 1.219rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

.c1 td:first-child,
.c1 th:first-child {
  border-radius: 8px 0 0 8px;
}

.c1 td:last-child,
.c1 th:last-child {
  border-radius: 0 8px 8px 0;
}

.c2 {
  padding: 1rem;
  position: relative;
  cursor: auto;
}

.c4 {
  padding: 1rem;
  position: relative;
  cursor: pointer;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
  white-space: nowrap;
  text-align: left;
  color: #00224B;
  font-size: 0.9375rem;
  font-weight: 500;
  line-height: 1.219rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

.c6 {
  width: 16px;
  height: 16px;
  -webkit-mask-size: 16px;
  mask-size: 16px;
  background: #00224B;
}

.c8 {
  padding: 1rem;
  text-align: left;
  border: 1px solid transparent;
  color: #00224B;
  font-size: 0.9375rem;
  font-weight: 400;
  line-height: 1.219rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

.c8:focus {
  border: 1px solid #8DC6FF;
  box-shadow: 0 0 0 4px #003FF23D;
}

.c7 {
  border-bottom: 1px solid rgba(0,0,0,0.08);
  display: table-row;
  background-color: #FFFFFF;
  border-top-left-radius: 10px;
}

@media (max-width:1024px) {
  .c0 {
    overflow-x: auto;
  }
}

<div
  class="c0"
  data-testid="DataTable"
>
  <table
    aria-rowcount="10"
    class="c1"
    role="grid"
  >
    <thead>
      <tr
        role="row"
      >
        <th
          class="c2"
          id="header-header1"
          role="columnheader"
        >
          <div
            class="c3"
          >
            Header 1
          </div>
        </th>
        <th
          class="c4"
          id="header-header2"
          role="columnheader"
        >
          <div
            class="c3"
          >
            Header 2
            <div
              aria-hidden="true"
              aria-label=""
              class="c5 c6"
              data-testid="Icon"
            />
          </div>
        </th>
        <th
          class="c4"
          id="header-header3"
          role="columnheader"
        >
          <div
            class="c3"
          >
            Header 3
            <div
              aria-hidden="true"
              aria-label=""
              class="c5 c6"
              data-testid="Icon"
            />
          </div>
        </th>
        <th
          class="c2"
          id="header-header4"
          role="columnheader"
        >
          <div
            class="c3"
          >
            Header 4
          </div>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        aria-rowindex="0"
        class="c7"
        role="row"
      >
        <td
          class="c8"
          headers="header-header1"
          role="cell"
          tabindex="0"
        >
          lorem ipsum
        </td>
        <td
          class="c8"
          headers="header-header2"
          role="cell"
          tabindex="-1"
        >
          R$ 1000
        </td>
        <td
          class="c8"
          headers="header-header3"
          role="cell"
          tabindex="-1"
        >
          Label
        </td>
        <td
          class="c8"
          headers="header-header4"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
      </tr>
      <tr
        aria-rowindex="1"
        class="c7"
        role="row"
      >
        <td
          class="c8"
          headers="header-header1"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8"
          headers="header-header2"
          role="cell"
          tabindex="-1"
        >
          R$ 1000
        </td>
        <td
          class="c8"
          headers="header-header3"
          role="cell"
          tabindex="-1"
        >
          Label
        </td>
        <td
          class="c8"
          headers="header-header4"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
      </tr>
      <tr
        aria-rowindex="2"
        class="c7"
        role="row"
      >
        <td
          class="c8"
          headers="header-header1"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8"
          headers="header-header2"
          role="cell"
          tabindex="-1"
        >
          R$ 1000
        </td>
        <td
          class="c8"
          headers="header-header3"
          role="cell"
          tabindex="-1"
        >
          Label
        </td>
        <td
          class="c8"
          headers="header-header4"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
      </tr>
      <tr
        aria-rowindex="3"
        class="c7"
        role="row"
      >
        <td
          class="c8"
          headers="header-header1"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8"
          headers="header-header2"
          role="cell"
          tabindex="-1"
        >
          R$ 1000
        </td>
        <td
          class="c8"
          headers="header-header3"
          role="cell"
          tabindex="-1"
        >
          Label
        </td>
        <td
          class="c8"
          headers="header-header4"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
      </tr>
      <tr
        aria-rowindex="4"
        class="c7"
        role="row"
      >
        <td
          class="c8"
          headers="header-header1"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8"
          headers="header-header2"
          role="cell"
          tabindex="-1"
        >
          R$ 1000
        </td>
        <td
          class="c8"
          headers="header-header3"
          role="cell"
          tabindex="-1"
        >
          Label
        </td>
        <td
          class="c8"
          headers="header-header4"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
      </tr>
      <tr
        aria-rowindex="5"
        class="c7"
        role="row"
      >
        <td
          class="c8"
          headers="header-header1"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8"
          headers="header-header2"
          role="cell"
          tabindex="-1"
        >
          R$ 1000
        </td>
        <td
          class="c8"
          headers="header-header3"
          role="cell"
          tabindex="-1"
        >
          Label
        </td>
        <td
          class="c8"
          headers="header-header4"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
      </tr>
      <tr
        aria-rowindex="6"
        class="c7"
        role="row"
      >
        <td
          class="c8"
          headers="header-header1"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8"
          headers="header-header2"
          role="cell"
          tabindex="-1"
        >
          R$ 1000
        </td>
        <td
          class="c8"
          headers="header-header3"
          role="cell"
          tabindex="-1"
        >
          Label
        </td>
        <td
          class="c8"
          headers="header-header4"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
      </tr>
      <tr
        aria-rowindex="7"
        class="c7"
        role="row"
      >
        <td
          class="c8"
          headers="header-header1"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8"
          headers="header-header2"
          role="cell"
          tabindex="-1"
        >
          R$ 1000
        </td>
        <td
          class="c8"
          headers="header-header3"
          role="cell"
          tabindex="-1"
        >
          Label
        </td>
        <td
          class="c8"
          headers="header-header4"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
      </tr>
      <tr
        aria-rowindex="8"
        class="c7"
        role="row"
      >
        <td
          class="c8"
          headers="header-header1"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8"
          headers="header-header2"
          role="cell"
          tabindex="-1"
        >
          R$ 1000
        </td>
        <td
          class="c8"
          headers="header-header3"
          role="cell"
          tabindex="-1"
        >
          Label
        </td>
        <td
          class="c8"
          headers="header-header4"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
      </tr>
      <tr
        aria-rowindex="9"
        class="c7"
        role="row"
      >
        <td
          class="c8"
          headers="header-header1"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
        <td
          class="c8"
          headers="header-header2"
          role="cell"
          tabindex="-1"
        >
          R$ 1000
        </td>
        <td
          class="c8"
          headers="header-header3"
          role="cell"
          tabindex="-1"
        >
          Label
        </td>
        <td
          class="c8"
          headers="header-header4"
          role="cell"
          tabindex="-1"
        >
          lorem ipsum
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
